src/integrationtests/java/com/aws/greengrass/integrationtests/e2e/BaseE2ETestCase.java

```diff
@@ -348,74 +396,117 @@ private static void createS3BucketsForTestComponentArtifacts() {
             ComponentIdentifier pkgIdLocal = getLocalPackageIdentifier(pkgId);
-            Path artifactDirPath = e2eTestPkgStoreDir.resolve("artifacts").resolve(pkgIdLocal.getName())
-                    .resolve(pkgIdLocal.getVersion().toString());
+            Path artifactDirPath =
+                    e2eTestPkgStoreDir
+                            .resolve("artifacts")
+                            .resolve(pkgIdLocal.getName())
+                            .resolve(pkgIdLocal.getVersion().toString());
             File[] artifactFiles = artifactDirPath.toFile().listFiles();
             if (artifactFiles == null) {
-                logger.atInfo().kv("component", pkgIdLocal).kv("artifactPath", artifactDirPath.toAbsolutePath())
+                logger.atInfo()
+                        .kv("component", pkgIdLocal)
+                        .kv("artifactPath", artifactDirPath.toAbsolutePath())
                         .log("Skip artifact upload. No artifacts found");
             } else {
                 for (File artifact : artifactFiles) {
                     try {
-                        // Path is <bucket>/<component_name>-<component>_<component_version>/<filename>
-                        s3Client.putObject(PutObjectRequest.builder().bucket(TEST_COMPONENT_ARTIFACTS_S3_BUCKET)
-                                .key(pkgIdLocal.getName() + "-" + pkgIdLocal.getVersion().toString() + "/" + artifact
-                                        .getName()).build(), RequestBody.fromFile(artifact));
+                        // Path is
+                        // <bucket>/<component_name>-<component>_<component_version>/<filename>
+                        s3Client.putObject(
+                                PutObjectRequest.builder()
+                                        .bucket(TEST_COMPONENT_ARTIFACTS_S3_BUCKET)
+                                        .key(
+                                                pkgIdLocal.getName()
+                                                        + "-"
+                                                        + pkgIdLocal.getVersion().toString()
+                                                        + "/"
+                                                        + artifact.getName())
+                                        .build(),
+                                RequestBody.fromFile(artifact));
                     } catch (S3Exception e) {
                         logger.atError().setCause(e).log("Could not upload artifacts to S3");
                     }
                 }
             }
-
         }
     }
 
     private static void cleanUpTestComponentArtifactsFromS3() {
         try {
-            ListObjectsResponse objectsInArtifactsBucket = s3Client.listObjects(
-                    ListObjectsRequest.builder().bucket(TEST_COMPONENT_ARTIFACTS_S3_BUCKET).build());
+            ListObjectsResponse objectsInArtifactsBucket =
+                    s3Client.listObjects(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.